pkg/node/manager/manager.go

```diff
@@ -464,7 +464,7 @@ func (m *manager) NodeUpdated(n nodeTypes.Node) {
 		// data such as for example the health endpoint.
 		existing := m.ipcache.GetMetadataByPrefix(prefix).Source()
 		overwrite := source.AllowOverwrite(existing, n.Source)
-		if !overwrite && existing != source.KubeAPIServer {
+		if !overwrite && existing != source.KubeAPIServer && address.Type != addressing.NodeCiliumInternalIP {
```

Comment:
I believe that the problem's root is in resolving the `exisitng` state of the node. When using `local-router-ipv4` all nodes will have the same internal IP address assigned (i.e. "169.254.4.6"). When a new node is added the code here incorrectly assumes that the Node is already populated because it already has an entry for IP "169.254.4.6".

What you are proposing is to effectively exclude NodeCiliumInternalIP from making the decision if NodeAdded event should be populated. This change generally LGTM.

nit: it will be more clear and easier to read if you move this check to the end of the loop and instead modyfing this condition use fail-fast construction like

```
if address.Type == addressing.NodeCiliumInternalIp {
    continue
}
```

Reply:
I don't know how the rest of this loop works, so I wanted to keep behavior changes to the absolute minimum. It would be great to have somebody that is more familiar with these code parts suggest better place to do this.

Reply:
Yeah, I'm not an expert on this as well. A comment from maintainers will be much appreciated here.

Reply:
I think I would agree with the code snippet shared in https://github.com/cilium/cilium/pull/28226/files#r1343917776. Additionally, I would suggest something like:

```go
if (option.Config.LocalRouterIPv4 == address || option.Config.LocalRouterIPv6 == address) && address.Type == addressing.NodeCiliumInternalIp {
    continue
}
```

so that the condition is never triggered unexpectedly.